src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,100 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elements;
+    private int size;
+
     @Override
     public void add(T value) {
-
+        if (elements == null) {
+            elements = (T[]) new Object[DEFAULT_CAPACITY];
+        }
+        if (size == elements.length) {
+            elements = Arrays.copyOf(elements, elements.length + elements.length / 2);
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index out of bounds");
+        }
+        if (elements == null) {
```

Comment:
Remove this check for null, as 'elements' should be initialized in the constructor.